sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp

```diff
@@ -557,6 +557,348 @@ inline __SYCL_ALWAYS_INLINE void joint_matrix_apply(
+  std::ignore = res;
+  std::ignore = src;
+  std::ignore = stride;
+  std::ignore = Layout;
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_ERROR_INVALID_DEVICE);
+#endif // defined(__SYCL_DEVICE_ONLY__)
+}
+
+template <
+    typename Group, typename S, typename T, use Use, size_t NumRows,
+    size_t NumCols, layout Layout, access::address_space Space,
+    access::decorated IsDecorated,
+    std::enable_if_t<std::is_same<S, std::remove_const_t<T>>::value ||
+                         (std::is_same<S, precision::tf32>::value &&
+                          std::is_same<std::remove_const_t<T>, float>::value),
+                     bool> = true>
+inline __SYCL_ALWAYS_INLINE void joint_matrix_load_checked(
+    Group sg, joint_matrix<Group, S, Use, NumRows, NumCols, Layout> &res,
+    multi_ptr<T, Space, IsDecorated> src, size_t stride, size_t Height,
+    size_t Width) {
+#if defined(__SYCL_DEVICE_ONLY__)
+  static_assert(Space != access::address_space::private_space,
+                "Joint Matrix doesn't support load from private memory!");
+  using DecorT = typename sycl::detail::DecoratedType<T, Space>::type;
+  DecorT *Ptr = sycl::detail::getDecorated<DecorT>(src);
+  res.spvm = __spirv_JointMatrixLoadCheckedINTEL<
+      DecorT, S, NumRows, NumCols, spv_matrix_use_traits<Use>::value,
+      spv_matrix_layout_traits<Layout>::value>(
+      Ptr, stride, Height, Width, spv_matrix_layout_traits<Layout>::value,
+      spv_scope_traits<Group>::value);
+#else
+  std::ignore = sg;
+  std::ignore = res;
+  std::ignore = src;
+  std::ignore = stride;
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_ERROR_INVALID_DEVICE);
+#endif // defined(__SYCL_DEVICE_ONLY__)
+}
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          access::address_space Space, access::decorated IsDecorated>
+inline __SYCL_ALWAYS_INLINE void
+joint_matrix_store_checked(Group sg,
+                           joint_matrix<Group, T, use::accumulator, NumRows,
+                                        NumCols, layout::dynamic> &src,
+                           multi_ptr<T, Space, IsDecorated> dst, size_t stride,
+                           layout Layout, size_t Height, size_t Width) {
+#if defined(__SYCL_DEVICE_ONLY__)
+  static_assert(Space != access::address_space::private_space,
+                "Joint Matrix doesn't support store to private memory!");
+  using DecorT = typename sycl::detail::DecoratedType<T, Space>::type;
+  DecorT *Ptr = sycl::detail::getDecorated<DecorT>(dst);
+  switch (Layout) {
+  default:
+    assert(false && "Invalid Memory Layout!");
+  case layout::row_major:
+    __spirv_JointMatrixStoreCheckedINTEL<
+        DecorT, T, NumRows, NumCols,
+        spv_matrix_use_traits<use::accumulator>::value,
+        spv_matrix_layout_traits<layout::dynamic>::value>(
+        Ptr, src.spvm, stride, Height, Width, __spv::MatrixLayout::RowMajor,
```

Comment:
Same question as above:
```suggestion
        Ptr, src.spvm, stride, Height, Width, spv_matrix_layout_traits<Layout>::value,
```

Reply:
No, it is not possible because layout in this case is a runtime variable.

Reply:
Please, see response in previous comment.